### PR TITLE
Add request conditional plugin

### PR DESCRIPTION
**Pull request by sagikazarmark**
Use case: cache certain responses based on a request condition.

### CODE REVIEW

File: src/RequestConditionalPlugin.php [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    /**
+     * @var Plugin
+     */
+    private $delegatedPlugin;
+
+    /**
+     * @param RequestMatcher $requestMatcher
+     * @param Plugin         $delegatedPlugin
+     */
+    public function __construct(RequestMatcher $requestMatcher, Plugin $delegatedPlugin)
```
Comment (Nyholm) [inline src/RequestConditionalPlugin.php:29]: I maybe want to use more than one plugin. I suggest allowing the second parameter to be an array of Plugins. Would that be possible?

Comment (sagikazarmark) [inline src/RequestConditionalPlugin.php:29]: I was also thinking about that. For that case I would add a separate
ChainPlugin.

Tobias Nyholm notifications@github.com ezt írta (2016. március 5.,
szombat):

Comment (Nyholm) [inline src/RequestConditionalPlugin.php:29]: Okey. Cool!

**Comment by sagikazarmark** [comment]
Anyone review?

File: src/RequestConditionalPlugin.php [hunk 2]
```diff
@@ -0,0 +1,46 @@
+<?php
+
+namespace Http\Client\Plugin;
+
+use Http\Message\RequestMatcher;
+use Psr\Http\Message\RequestInterface;
+
+/**
+ * Apply a delegated plugin based on a request match.
```
Comment (dbu) [inline src/RequestConditionalPlugin.php:9]: Execute a plugin only if the RequestMatcher matches the request.

File: src/RequestConditionalPlugin.php [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+ *
+ * @author Márk Sági-Kazár <mark.sagikazar@gmail.com>
+ */
+final class RequestConditionalPlugin implements Plugin
```
Comment (dbu) [inline src/RequestConditionalPlugin.php:13]: i am not too happy with the name. the name does not really tell how the 3 terms go together. is this a plugin to request a conditional? or to make the request conditional?

but unfortunately i have no better name. ConditionalRequestPlugin certainly is worse.

**Comment by dbu** [comment]
would this now need to go against client-commons, now that we move the plugins there?

File: src/RequestConditionalPlugin.php (hunk 3 above)
Comment (sagikazarmark) [inline src/RequestConditionalPlugin.php:13]: Alternative could be RequestMatcherPlugin. It's kinda obvious, don't know why I didn't think about it earlier.

**Comment by sagikazarmark** [comment]
Yes, it should. Wanted to finish it before closing and moving to common.

File: src/RequestConditionalPlugin.php (hunk 3 above)
Comment (dbu) [inline src/RequestConditionalPlugin.php:13]: yep, RequestMatcherPlugin is a good name!

**Pull request closed without merging**